core/src/main/java/org/owasp/dependencycheck/analyzer/AbstractSuppressionAnalyzer.java

```diff
@@ -113,9 +122,24 @@ protected void analyzeDependency(Dependency dependency, Engine engine) throws An
         if (rules.isEmpty()) {
             return;
         }
-        rules.forEach((rule) -> rule.process(dependency));
+        int ctr = 0;
+        for (SuppressionRule rule : rules.list()) {
+            if (filter(rule)) {
+                ctr++;
+                rule.process(dependency);
+            }
```

Comment:
A left-over counter from initial debugging? It is only ever written to, never used.

```suggestion
        for (SuppressionRule rule : rules.list()) {
            if (filter(rule)) {
                rule.process(dependency);
            }
```